sources/osgAnimation/UpdateMorph.js

```diff
@@ -47,21 +56,174 @@ define( [
             return this._isInitialized;
         },
 
-        getTarget: function ( index ) {
-            return this._targets[ index ];
+        getNumTarget: function () {
+            return this._targets.length;
         },
 
-        getNumTarget: function () {
-            return this._targetName.length;
+        getTarget: function ( index ) {
+            return this._targets[ index ];
         },
 
         getTargetName: function ( index ) {
-            return this._targetName[ index ];
+            return this._targetNames[ index ];
         },
 
         addTarget: function ( name, index ) {
             this._targets[ index ] = Target.createFloatTarget( 0 );
-            this._targetName[ index ] = name;
+            this._targetNames[ index ] = name;
+        },
+
+        _remapBufferArrays: function () {
+
+            // basically, this function remaps all the active morphed VA to the 4th first morphTargets VA
+            var indexMap = this._indexMap;
+            var morphs = this._morphs;
+            for ( var i = 0; i < 4; ++i ) {
+                var index = indexMap[ i ];
+
+                for ( var j = 0, nbMorphs = morphs.length; j < nbMorphs; ++j ) {
+
+                    var morph = morphs[ j ];
+                    var vAttrs = morph.getVertexAttributeList();
+                    var morphNames = morph.getMorphTargetNames();
+                    for ( var k = 0, nbNames = morphNames.length; k < nbNames; ++k ) {
+
+                        var attName = morphNames[ k ];
+                        vAttrs[ attName + '_' + i ].setActiveArray( vAttrs[ attName + '_' + index ] );
+
+                    }
+                }
+            }
+        },
+
+        _mergeExtraMorphTarget: function ( attrs, attName, extraWeightSum ) {
+
+            var i = 0;
+            // ignore the gpu morphed
+            var gpuMorphed = this._gpuMorphed;
+            var vAttr = attrs[ attName ];
+            var vertexLen = vAttr.getElements().length;
+            var nbVertex = vertexLen / vAttr.getItemSize();
+            var itemSize = vAttr.getItemSize();
+
+            vAttr._cpuMorph = vAttr._cpuMorph || new BufferArray( BufferArray.ARRAY_BUFFER, new Float32Array( vertexLen ), itemSize );
+            var morphExtraTargets = vAttr._cpuMorph.getElements();
+            for ( i = 0; i < vertexLen; ++i ) {
+                morphExtraTargets[ i ] = 0.0;
+            }
+
+            var targets = this._targets;
+            for ( var j = 0, nb = targets.length; j < nb; ++j ) {
+                var weight = targets[ j ].value;
+                if ( gpuMorphed[ j ] === true ) continue; // ignore gpu morphed targets
+                if ( Math.abs( weight ) < WEIGHT_CPU_EPS ) continue;
+
+                weight /= extraWeightSum;
```

Comment:
It can be "near 0" but always >= WEIGHT_CPU_EPS 
the same code is used to compute the extra weight https://github.com/cedricpinson/osgjs/blob/morph-many-targets/sources/osgAnimation/UpdateMorph.js#L145-L147

si it should be safe